app/services/vk-feed.ts

```diff
@@ -1,31 +1,19 @@
 import { appProvider } from './app'
-import { Injectable, Component, ComponentFactory, ComponentRef, Compiler } from '@angular/core'
-
-@Component({
-    selector: 'vk-feed-widget',
-    template: `<div id="vk_feed_{{ID}}"></div>`
-})
-
-class VkFeedWidget {
-    public ID :number;
-}
+import { Injectable, ComponentFactoryResolver, ComponentRef } from '@angular/core'
+import { VkFeedWidget } from '../components.common/vk-feed/vk-feed-widget'
 
 declare var VK :any;
 
 @Injectable()
 export class VkFeedService {
     private widgets = {};
 
-    private sharedComponent :any;
-
-    constructor (compiler :Compiler) {
-        this.sharedComponent = compiler.compileComponentAsync(VkFeedWidget);
-    }
+    constructor(private componentResolver :ComponentFactoryResolver) { }
 
     private makeWidget (params :{ id :number, options ?:any}, key: string) {
-        return this
-            .sharedComponent
-            .then((factory :ComponentFactory<any>) => appProvider.app.instance.viewRef.createComponent(factory))
+        let vkFeedWidgetFactory = this.componentResolver.resolveComponentFactory(VkFeedWidget);
```

Comment:
Check:
* `VkFeedWidget` must be in `entryComponent` list
* `makeWidget` must be called in `AfterContentInit` cycle